contracts/lib/GenesisLib.sol

```diff
@@ -13,8 +13,9 @@ enum GenesisStatus {
 }
 
 struct GenesisValidator {
-    address validator;
+    address addr;
```

Comment:
@ZeroEkkusu Is this change safe to be made? The expectation for this PR is that the chain needs to be started from the genesis in order to be able to use the premine functionality however, I'm wondering what would happen for the already running chains when they apply this change and is it ok to rename it or revert it just in case to be backward compatible. WDYT?

Reply:
WDYT about such change? I've introduced an array which stores premine balances and exposed a function that retrieves it (because we are going to need it on the client-side, it is easier that way than having mapping and fetching balances by `address` key, one by one):
https://github.com/0xPolygon/core-contracts/pull/411/commits/9f63d588ab6d30cf8cf9046c8683de9352acb08f

Reply:
Re your comment/post at the end:

Do you mean that a restart will be required  anyway? If so, different storage layouts don't matter because those would be fresh deployments with clean state.

If not:

This also shifts the storage layout of `CustomSupernetManager` (because there's `GenesisSet private _genesis` in it).

We can have the array defined and updated in the `CustomSupernetManager` directly (after all other storage variables and 1 subtracted from the gap), if that works for you.

Reply:
> Do you mean that a restart will be required anyway?

It will be required in case one uses rootchain originated token as a native token. Otherwise we don't expect chain reset (redeployment is not expected in that case).

Reply:
In that case we'll need to align the storage layouts. Does the solution I proposed work for you?

Reply:
> We can have the array defined and updated in the CustomSupernetManager directly (after all other storage variables and 1 subtracted from the gap), if that works for you.

Yes, that should work. I'll change it that way most likely. :+1: